applicationset/generators/merge.go

```diff
@@ -41,7 +41,7 @@ func (m *MergeGenerator) getParamSetsForAllGenerators(generators []argoprojiov1a
 	for _, generator := range generators {
 		generatorParamSets, err := m.getParams(generator, appSet)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error getting params from generator: %w", err)
```

Comment:
How about getting just slightly fancier?

```suggestion
			return nil, fmt.Errorf("error getting params from generator %d of %d: %w", i+1, len(generators), err)
```

Reply:
makes sense, its definitely better to give more context

Reply:
Yep!